extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java

```diff
@@ -0,0 +1,119 @@
+package io.quarkus.smallrye.reactivemessaging.runtime;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.BeforeDestroyed;
+import javax.enterprise.event.Observes;
+import javax.enterprise.event.Reception;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.arc.AlternativePriority;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.reactive.messaging.annotations.Blocking;
+import io.smallrye.reactive.messaging.connectors.ExecutionHolder;
+import io.smallrye.reactive.messaging.connectors.WorkerPoolRegistry;
+import io.smallrye.reactive.messaging.helpers.Validation;
+import io.vertx.core.Handler;
+import io.vertx.mutiny.core.Promise;
+import io.vertx.mutiny.core.WorkerExecutor;
+
+@AlternativePriority(1)
+@ApplicationScoped
+// TODO: create a different entry for WorkerPoolRegistry than `analyzeWorker` and drop this class
+public class QuarkusWorkerPoolRegistry extends WorkerPoolRegistry {
```

Comment:
IS the only difference with the upstream version the AlternativePriority?

Reply:
the difference is the `defineWorker` method; the upstream version only works on `AnnotatedType`: https://github.com/smallrye/smallrye-reactive-messaging/blob/master/smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/connectors/WorkerPoolRegistry.java#L98

Reply:
Thanks!